beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -212,6 +240,35 @@ impl<E: EthSpec> PendingComponents<E> {
             AvailableExecutedBlock::new(available_block, import_data, payload_verification_outcome),
         )))
     }
+
+    /// Returns the epoch of the block if it is cached, otherwise returns the epoch of the first blob.
+    pub fn epoch(&self) -> Option<Epoch> {
```

Comment:
Maybe we just pass the epoch to the da_checker functions instead of trying to calculate it every time?

Reply:
Not sure if I fully understand, we would still have to calculate the epoch from the slot of the block / blob right? or do you mean storing the `epoch` in `PendingComponents` whenever the first component is added so we don't re-calculate?

Reply:
Added in [379f40a](https://github.com/sigp/lighthouse/pull/6076/commits/379f40a345a4276f836f9bc24ef8e75b68b54056)